### PR TITLE
[Draft] Add ConfigMap for LimeSurvey ldap.php

### PR DESCRIPTION
**Pull request by martialblog**
First draft fo a ConfigMap to configure the LimeSurvey ldap.php.

**TODO**

- [x]  Extend Tests
- [x]  Add Example
- [x]  Update Readme

Not sure if `.Values.ldapConfigMap ` is the right place and name. 

Feedback is welcome.

### CODE REVIEW

**Comment by martialblog** [comment]
See also #21

File: limesurvey/templates/deployment.yaml [hunk 1]
```diff
@@ -73,6 +73,11 @@ spec:
               {{- if .Values.persistence.subPath }}
               subPath: {{ .Values.persistence.subPath }}
               {{- end }}
+          {{- if .Values.ldapConfigMap }}
+            - name: ldap-config
+              mountPath: /varl/www/html/application/config/ldap.php
```
Comment (jacksgt) [inline limesurvey/templates/deployment.yaml:78]: Typo `varl` -> `var`

File: limesurvey/templates/deployment.yaml [hunk 2]
```diff
@@ -197,6 +202,14 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       volumes:
+      {{- if .Values.ldapConfigMap }}
+        - name: ldap-config
+          configMap:
+            name: {{ include "limesurvey.ldapConfigMapName" . }}
+            items:
+            - key: "ldap.php"
```
Comment (jacksgt) [inline limesurvey/templates/deployment.yaml:210]: Not sure if I understand the docs correctly, but won't this mount the ldap.php to `keys/ldap.php`?
https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/#KeyToPath

**Review by jacksgt — CHANGES_REQUESTED** [review]

File: limesurvey/templates/deployment.yaml (hunk 1 above)
Comment (martialblog) [reply limesurvey/templates/deployment.yaml]: Fixed

File: limesurvey/templates/deployment.yaml (hunk 2 above)
Comment (martialblog) [reply limesurvey/templates/deployment.yaml]: Think I misunderstood that. I just pushed some code that I tested.

**Comment by martialblog** [comment]
I think it makes more sense to have a generic way to pass extra Files into the Template. Back to the drawing board

**Pull request closed without merging**